app/vmalert/notifier/alertmanager_request.qtpl

```diff
@@ -15,10 +15,10 @@
     "endsAt":{%q= alert.End.Format(time.RFC3339Nano) %},
     {% endif %}
     "labels": {
-        "alertname":{%q= alert.Name %}
```

Comment:
Hm, this one is tricky. 
According to the [spec](https://github.com/prometheus/compliance/blob/main/alert_generator/specification.md#alert-format)  `alertname` needs to be set explicitly.
However, in Prometheus they don't do it explicitly. They just use labels as you do in this PR. Which makes `alertname` vulnerable during relabeling. It means the user can drop this label via relabeling.

I thought this would lead to an error on the Alertmanager side. But I didn't find any evidence of that. It looks like `alertname` is optional. This is weird.

I assume, your approach is correct...

Reply:
yeah, it's weird